benchmarks/graphics_pipeline/main.cpp

```diff
@@ -434,71 +489,83 @@ void ProjApp::Setup()
-            uint32_t y     = index % grid.ySize;
-            uint32_t z     = index / (grid.xSize * grid.ySize);
-
-            // Model matrix to be applied to the sphere mesh
-            float4x4 modelMatrix = glm::translate(float3(x * grid.step, y * grid.step, z * grid.step));
-
-            // Copy a sphere mesh to create a giant vertex buffer
-            // Iterate through the meshes vertx data and add it to the geometry
-            for (uint32_t vertexIndex = 0; vertexIndex < sphereVertexCount; ++vertexIndex) {
-                TriMeshVertexData vertexData = {};
-                mesh.GetVertexData(vertexIndex, &vertexData);
-                vertexData.position = modelMatrix * float4(vertexData.position, 1);
-
-                TriMeshVertexDataCompressed vertexDataCompressed;
-                vertexDataCompressed.position = half3(glm::packHalf1x16(vertexData.position.x), glm::packHalf1x16(vertexData.position.y), glm::packHalf1x16(vertexData.position.z));
-                vertexDataCompressed.texCoord = half2(glm::packHalf1x16(vertexData.texCoord.x), glm::packHalf1x16(vertexData.texCoord.y));
-                vertexDataCompressed.normal   = i8vec4(MapFloatToInt8(vertexData.normal.x), MapFloatToInt8(vertexData.normal.y), MapFloatToInt8(vertexData.normal.z), MapFloatToInt8(1.0f));
-                vertexDataCompressed.tangent  = i8vec4(MapFloatToInt8(vertexData.tangent.x), MapFloatToInt8(vertexData.tangent.y), MapFloatToInt8(vertexData.tangent.z), MapFloatToInt8(vertexData.tangent.a));
-                lowPrecisionInterleaved.AppendVertexData(vertexDataCompressed);
-                lowPrecisionPositionPlanar.AppendVertexData(vertexDataCompressed);
-
-                highPrecisionInterleaved.AppendVertexData(vertexData);
-                highPrecisionPositionPlanar.AppendVertexData(vertexData);
-            }
-            // Iterate the meshes triangles and add the vertex indices
-            for (uint32_t triIndex = 0; triIndex < sphereTriCount; ++triIndex) {
-                uint32_t v0 = PPX_VALUE_IGNORED;
-                uint32_t v1 = PPX_VALUE_IGNORED;
-                uint32_t v2 = PPX_VALUE_IGNORED;
-                mesh.GetTriangle(triIndex, v0, v1, v2);
-                lowPrecisionInterleaved.AppendIndicesTriangle(v0 + i * sphereVertexCount, v1 + i * sphereVertexCount, v2 + i * sphereVertexCount);
-                lowPrecisionPositionPlanar.AppendIndicesTriangle(v0 + i * sphereVertexCount, v1 + i * sphereVertexCount, v2 + i * sphereVertexCount);
-                highPrecisionInterleaved.AppendIndicesTriangle(v0 + i * sphereVertexCount, v1 + i * sphereVertexCount, v2 + i * sphereVertexCount);
-                highPrecisionPositionPlanar.AppendIndicesTriangle(v0 + i * sphereVertexCount, v1 + i * sphereVertexCount, v2 + i * sphereVertexCount);
+        // LODs for spheres
+        mSphereLODs.push_back(LOD{/* longitudeSegments = */ 10, /* latitudeSegments = */ 10, kAvailableLODs[0]});
```

Comment:
Instead of having fixed sized LODs, could the segment count be a slider?
Every time you update the slider you'll have to recreate the meshes, but you'd be more flexible in choosing the subdivision level.

Reply:
The goal for this knob here is to make the benchmark either bound by vertex input/shading (high vertex count) or bound by other stages (low vertex count), so I think it would be fine to have just fixed sized LOD for now.  We can probably push this version first, and add the segment count in the future if we find that it is required for some other tests. :)

Reply:
Sure, okok